src/devices/machine/z80scc.cpp

```diff
@@ -2893,9 +2907,17 @@ void z80scc_channel::update_serial()
 	}
 	else
 	{
-		LOG("- BRG disabled\n");
-		set_rcv_rate(0);
-		set_tra_rate(0);
+		if ((m_wr11 & WR11_RCVCLK_SRC_MASK) == WR11_RCVCLK_SRC_RTXC
+				&& (m_wr11 & WR11_TRACLK_SRC_MASK) == WR11_TRACLK_SRC_RTXC) {
+			m_brg_rate = get_rtxc_rate();
+			LOG("- BRG disabled, clock source RTxC (rate %d, clock %d)\n", m_brg_rate, get_clock_mode());
+			set_rcv_rate(m_brg_rate);
+			set_tra_rate(m_brg_rate);
+		} else {
+			LOG("- BRG disabled and RX/TX clock sources differ, unimplemented: stopping\n");
+			set_rcv_rate(0);
+			set_tra_rate(0);
+		}
```

Comment:
Please follow the formatting of the file you’re editing.  The rest of this file uses Allman block formatting, don’t use K&R formatting for new code.